cunumeric/array.py

```diff
@@ -150,7 +153,8 @@ def convert_to_cunumeric_ndarray(obj: Any, share: bool = False) -> ndarray:
         return obj
     # Ask the runtime to make a numpy thunk for this object
     thunk = runtime.get_numpy_thunk(obj, share=share)
-    return ndarray(shape=None, thunk=thunk)
+    flags = obj.flags if isinstance(obj, np.ndarray) and share else None
+    return ndarray(shape=None, thunk=thunk, flags=flags)
```

Comment:
Let's also propagate the other way around, i.e. when creating a NumPy array from a cuNumeric array. I suggest doing something like the following in `def __array__()`:

```
        numpy_array = self._thunk.__numpy_array__()
        numpy_array.flags.writeable = self._writeable
        if dtype is not None:
            numpy_array = numpy_array.astype(dtype)
        return numpy_array
```

Reply:
```
   >       numpy_array.flags.writeable = self._writeable
   E       ValueError: cannot set WRITEABLE flag to True of this array
```

Reply:
Hrm, that's curious. Do you have a minimal reproducer?

Reply:
tests/integration/test_random_straightforward.py

Reply:
Looks like NumPy doesn't like setting `writeable=True` when `owndata` is `False`, even if `writeable` is already `True`...

We should be able to always downgrade `writeable` to `False`, so maybe this is the best we can do

```
numpy_array = self._thunk.__numpy_array__()
if numpy_array.flags.writeable and not self._writeable:
    numpy_array.flags.writeable = False
if dtype is not None:
    numpy_array = numpy_array.astype(dtype)
return numpy_array
```

Reply:
https://github.com/nv-legate/cunumeric/pull/759/commits/0fd07f52df082a966cf9ca2cfe82342c7e7fcdc2